src/pages/Projects/Projects/components/ProjectModal/ProjectModal.tsx

```diff
@@ -28,79 +28,94 @@ import { useForm } from "react-hook-form";
+  addProject?: (project: NewProject) => void;
+  updateProject?: (id: string, updated: Partial<Project>) => void;
   project?: Project;
 }
 
 type SaveButtonStatus = "idle" | "pending";
 
-export default function AddProjectModal({
+export default function ProjectModal({
   isOpen,
   onClose,
   addProject,
   project,
-}: AddProjectModalProps): JSX.Element {
+  updateProject,
+}: ProjectModalProps): JSX.Element {
   const history = useHistory();
   const [serverError, setServerError] = useState(false);
   const [status, setStatus] = useState<SaveButtonStatus>("idle");
-  const [newProject, setNewProject] = useState<FormData>(initialFormState);
 
   const {
     register,
-    watch,
     handleSubmit,
     reset,
-    formState: { errors, isDirty: formIsDirty },
+    setValue,
+    watch,
+    getValues,
+    formState: { errors, isDirty },
   } = useForm<ProjectFormData>({
     defaultValues: project ? project : undefined,
   });
-
-  const isNewProject = isEmpty(project);
   const projectName = watch("name");
-  const canSubmitForm =
-    (isNewProject && fullNameProvided(projectName)) ||
-    (!isNewProject && formIsDirty && fullNameProvided(projectName));
-
+  const isNewProject = isEmpty(project);
   const addNewProject = async () => {
-    try {
-      setStatus("pending");
-      const newProjectId = await addProject(newProject);
-      history.push(`/projects/${newProjectId}`);
-      resetForm();
-      onClose();
-      setStatus("idle");
-    } catch (e) {
-      setServerError(!serverError);
+    if (addProject) {
+      try {
+        setStatus("pending");
+        const desc = getValues("description");
+        const newProjectId = await addProject({
+          name: projectName,
+          description: desc,
+        });
+        history.push(`/projects/${newProjectId}`);
+        resetForm();
+        onClose();
+        setStatus("idle");
+        setServerError(false);
+      } catch (e) {
+        setServerError(true);
+      }
     }
   };
 
-  const handleChange = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
-  ) => {
-    const { value, name } = e.target;
-    setNewProject((newProject) => ({
-      ...newProject,
-      [name]: value,
-    }));
+  const update = async () => {
+    if (project && updateProject && isDirty) {
+      try {
+        setStatus("pending");
+        const desc = getValues("description");
+        updateProject(project.id, { name: projectName, description: desc });
+        resetForm();
+        onClose();
+        setStatus("idle");
+        setServerError(false);
+      } catch (e) {
+        setServerError(true);
+      }
+    }
   };
 
   const resetForm = () => {
-    reset({ name: "", description: "" });
-    setNewProject(initialFormState);
+    reset({
+      name: project ? project.name : "",
+      description: project ? project.description : "",
+    });
   };
 
+  useEffect(() => {
+    if (project) {
+      setValue("name", project?.name);
```

Comment:
there is no need for the optional chaining as you're already checking if project exists